### PR TITLE
New Formula pcp

### PR DESCRIPTION
**Pull request by parthbeingparth**
- [ Yes ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [ Yes ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ Yes ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ Yes ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/pcp.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+class Pcp < Formula
+  desc "Performance Co-Pilot is a system performance and analysis framework"
+  homepage "https://pcp.io/"
+  url "https://dl.bintray.com/pcp/source/pcp-4.1.3.src.tar.gz"
```
Comment (fxcoudert) [inline Formula/pcp.rb]: Please use the github release tarballs, if possible, as this allows us to automatically check for updates

File: Formula/pcp.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  depends_on "pkg-config" => :build
+  depends_on "nss"
+  depends_on "python"
+  depends_on "qt"
+  depends_on "readline"
+  depends_on "xz"
+
+  def install
+    system "./configure", "--prefix=#{prefix}", "CC=clang", "CXX=clang", "--without-manager"
```
Comment (fxcoudert) [inline Formula/pcp.rb]: Doesn't `configure` pick up the `CC` and `CXX` values from environment?
(And if it doesn't: shouldn't `CXX` be `clang++`?)

Comment (fxcoudert) [inline Formula/pcp.rb]: Why build `--without-manager` if it is not the default?

File: Formula/pcp.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  end
+
+  plist_options :manual => "pcp"
+
+  def plist; <<~EOS
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+    <dict>
+        <key>RunAtLoad</key>         <true/>
```
Comment (fxcoudert) [inline Formula/pcp.rb]: No spacing here. Key on the next line.

File: Formula/pcp.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        <key>KeepAlive</key>         <true/>
+        <key>Label</key>             <string>io.pcp.startup</string>
+
```
Comment (fxcoudert) [inline Formula/pcp.rb]: No blank lines.

File: Formula/pcp.rb [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        <key>ProgramArguments</key>
+            <array>
+              <string>ln</string>
+                <string>-s</string>
```
Comment (fxcoudert) [inline Formula/pcp.rb]: Too much indentation, all array elements should be at same level

File: Formula/pcp.rb [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+                <string>/etc/pcp.conf</string>
+            </array>
+
+    </dict>
+    </plist>
+  EOS
+  end
+
+  test do
+    system "/usr/local/Cellar/pcp/4.1.1/etc/init.d/pmcd", "start"
```
Comment (fxcoudert) [inline Formula/pcp.rb]: Use `#{etc}/init.d/pmcd`. Can we check the output?

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/pcp.rb (hunk 1 above)
Comment (Warewolf96) [reply Formula/pcp.rb]: Resolved this...

File: Formula/pcp.rb (hunk 2 above)
Comment (Warewolf96) [reply Formula/pcp.rb]: We were using this for initial testing but the formula works without it, updated in the latest commit.

Comment (Warewolf96) [reply Formula/pcp.rb]: We were using this for initial testing but the formula works without it, updated in the latest commit.

File: Formula/pcp.rb (hunk 3 above)
Comment (Warewolf96) [reply Formula/pcp.rb]: There should be a blank line before key?

File: Formula/pcp.rb (hunk 5 above)
Comment (Warewolf96) [reply Formula/pcp.rb]: Rectified.

File: Formula/pcp.rb (hunk 4 above)
Comment (Warewolf96) [reply Formula/pcp.rb]: Rectified.

File: Formula/pcp.rb (hunk 6 above)
Comment (Warewolf96) [reply Formula/pcp.rb]: The output will be pmcd successfully running.

**Comment by Warewolf96** [comment]
@fxcoudert Sir, Can you please review the latest commit?

**Comment by fxcoudert** [comment]
This is currently not building:

```
==> ./configure --prefix=/usr/local/Cellar/pcp/4.1.3
checking build system type... x86_64-apple-darwin18.0.0
checking host system type... x86_64-apple-darwin18.0.0
checking target system type... x86_64-apple-darwin18.0.0
Building on x86_64-apple-darwin18.0.0 for x86_64-apple-darwin18.0.0
Build: os=darwin18.0.0 cpu=x86_64
Target: os=darwin cpu=x86_64
checking for pkg-config... /usr/local/opt/pkg-config/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking whether the C compiler works... no
configure: error: in `/private/tmp/pcp-20180927-6928-1g78ua8/pcp-4.1.3':
configure: error: C compiler cannot create executables
See `config.log' for more details
```

The review comments about the plist formatting have not been addressed.

**Comment by fxcoudert** [comment]
@BrewTestBot test this please

**Comment by fxcoudert** [comment]
Same error… something is clearly very wrong with the configure script. I am closing the pull request for now, please consider reopening once the formula builds fine locally (and audit passes).

**Pull request closed without merging**